app/routes/info.py

```
@@ -843,6 +857,49 @@ async def get_z3ds_info(
+        )
+
+    try:
+        info = await run_in_threadpool(nkit2iso_service.info, path)
+    except NkitHeaderError as e:
+        # The name says .nkit.iso but the bytes disagree — a client error about
+        # the file, not a server fault, so 422 rather than 500.
+        raise HTTPException(status_code=422, detail=str(e)) from None
+    except Exception as e:
+        # Broad by design at the file-read boundary (OSError, struct/zlib
+        # errors, ...); logged here so the 500 is traceable.
+        logger.exception("Failed to read NKit info for %s: %s", path, e)
```

Ruff (0.16.1)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)